src/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.Proxy.cs

```diff
@@ -0,0 +1,302 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.Net.Test.Common;
+using System.Text;
+using System.Threading.Tasks;
+
+using Xunit;
+using Xunit.Abstractions;
+
+namespace System.Net.Http.Functional.Tests
+{
+    using Configuration = System.Net.Test.Common.Configuration;
+
+    [SkipOnTargetFramework(TargetFrameworkMonikers.Uap, "UAP HTTP stack doesn't support .Proxy property")]
+    public abstract class HttpClientHandler_Proxy_Test : HttpClientTestBase
+    {
+        private readonly ITestOutputHelper _output;
+        
+        public HttpClientHandler_Proxy_Test(ITestOutputHelper output)
+        {
+            _output = output;
+        }
+        
+        [OuterLoop("Uses external server")]
+        [Theory]
+        //[ActiveIssue(30327)]
+        //[InlineData(AuthenticationSchemes.Ntlm, false, true)]
+        //[InlineData(AuthenticationSchemes.Negotiate, false, true)]
+        //[InlineData(AuthenticationSchemes.Ntlm, true, true)]
+        //[InlineData(AuthenticationSchemes.Negotiate, true, true)]
+
+        [InlineData(AuthenticationSchemes.Ntlm, true, false)]
+        [InlineData(AuthenticationSchemes.Negotiate, true, false)]
+        [InlineData(AuthenticationSchemes.Basic, false, false)]
+        [InlineData(AuthenticationSchemes.Basic, true, false)]
+        [InlineData(AuthenticationSchemes.Digest, false, false)]
+        [InlineData(AuthenticationSchemes.Digest, true, false)]
+        [InlineData(AuthenticationSchemes.Ntlm, false, false)]
+        [InlineData(AuthenticationSchemes.Negotiate, false, false)]
+        [InlineData(AuthenticationSchemes.Basic, false, true)]
+        [InlineData(AuthenticationSchemes.Basic, true, true)]
+        [InlineData(AuthenticationSchemes.Digest, false, true)]
+        [InlineData(AuthenticationSchemes.Digest, true, true)]
+        public async Task AuthProxy__ValidCreds_ProxySendsRequestToServer(
+            AuthenticationSchemes proxyAuthScheme,
+            bool secureServer,
+            bool proxyClosesConnectionAfterFirst407Response)
+        {
+            if (PlatformDetection.IsWindowsNanoServer && IsWinHttpHandler && proxyAuthScheme == AuthenticationSchemes.Digest)
+            {
+                // WinHTTP doesn't support Digest very well and is disabled on Nano.
+                return;
+            }
+
+            if (PlatformDetection.IsFullFramework &&
+                (proxyAuthScheme == AuthenticationSchemes.Negotiate || proxyAuthScheme == AuthenticationSchemes.Ntlm))
+            {
+                // Skip Windows auth tests on .NET Framework.
```

Comment:
Why do we skip them on netfx?

Reply:
I'm getting weird errors on NETFX and I think it is due to a bug in NETFX.  I didn't think it was worth tracking a NETFX issue for that so I'm skipping the tests on NETFX.

Reply:
Ok.  My only hesitation would be if it's not a netfx bug but rather a test bug, and we're ending up testing the wrong thing on the other handlers.  If it's definitely not a test issue, this is fine, though it'd be nice to add a comment about why it's being skipped on netfx so that the next reader who comes along doesn't have the same question :)

Reply:
It looks like a NETFX bug.  I broke into the debugger and looked at the call stack and source code. I'll open up a separate NETFX bug in the .NET Framework bug database. I'll update the comments to explain why the test is being skipped on NETFX.

```
System.IO.IOException
  HResult=0x80131620
  Message=The operation is not allowed on non-connected sockets.
  Source=System
  StackTrace:
   at System.Net.Sockets.NetworkStream.InitNetworkStream(Socket socket, FileAccess Access) in line 409
System.dll!System.Net.Sockets.NetworkStream.InitNetworkStream(System.Net.Sockets.Socket socket, System.IO.FileAccess Access) Line 409	
System.dll!System.Net.Sockets.NetworkStream.NetworkStream(System.Net.Sockets.NetworkStream networkStream, bool ownsSocket) Line 90	
System.dll!System.Net.Connection.TunnelThroughProxyWrapper(System.IAsyncResult result) Line 3664	
System.dll!System.Net.LazyAsyncResult.Complete(System.IntPtr userToken) Line 415	
mscorlib.dll!System.Threading.ExecutionContext.RunInternal(System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, object state, bool preserveSyncCtx) Line 954	
mscorlib.dll!System.Threading.ExecutionContext.Run(System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, object state, bool preserveSyncCtx) Line 902	
mscorlib.dll!System.Threading.ExecutionContext.Run(System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, object state) Line 891	
System.dll!System.Net.ContextAwareResult.Complete(System.IntPtr userToken) Line 465	
System.dll!System.Net.LazyAsyncResult.ProtectedInvokeCallback(object result, System.IntPtr userToken) Line 370	
System.dll!System.Net.HttpWebRequest.ProcessResponse() Line 3789	
System.dll!System.Net.HttpWebRequest.SetResponse(System.Net.CoreResponseData coreResponseData) Line 3730	
System.dll!System.Net.ConnectionReturnResult.SetResponses(System.Net.ConnectionReturnResult returnResult) Line 200	
System.dll!System.Net.Connection.ReadComplete(int bytesRead, System.Net.WebExceptionStatus errorStatus) Line 3546	
System.dll!System.Net.LazyAsyncResult.Complete(System.IntPtr userToken) Line 415	
System.dll!System.Net.ContextAwareResult.Complete(System.IntPtr userToken) Line 465	
System.dll!System.Net.LazyAsyncResult.ProtectedInvokeCallback(object result, System.IntPtr userToken) Line 370	
System.dll!System.Net.Sockets.BaseOverlappedAsyncResult.CompletionPortCallback(uint errorCode, uint numBytes, System.Threading.NativeOverlapped* nativeOverlapped) Line 399	
mscorlib.dll!System.Threading._IOCompletionCallback.PerformIOCompletionCallback(uint errorCode, uint numBytes, System.Threading.NativeOverlapped* pOVERLAP) Line 135	
[Native to Managed Transition]	
```